apps/studio/components/interfaces/SQLEditor/DownloadSnippetModal.tsx

```diff
@@ -0,0 +1,150 @@
+import { ModalProps } from '@ui/components/Modal/Modal'
+import { snakeCase } from 'lodash'
+import { useSqlEditorStateSnapshot } from 'state/sql-editor'
+import { CodeBlock, Modal, Separator, Tabs, TabsProvider } from 'ui'
+import { useState } from 'react'
+import {
+  generateFileCliCommand,
+  generateMigrationCliCommand,
+  generateSeedCliCommand,
+} from './SQLEditor.utils'
+import TwoOptionToggle from 'components/ui/TwoOptionToggle'
+
+export interface DownloadSnippetModalProps extends ModalProps {
+  id: string
+}
+
+const DownloadSnippetModal = ({ id, ...props }: DownloadSnippetModalProps) => {
+  const snap = useSqlEditorStateSnapshot()
+  const snippet = snap.snippets[id].snippet
+  const migrationName = snakeCase(snippet.name)
+
+  const [selectedView, setSelectedView] = useState<'CLI' | 'NPM'>('CLI')
+
+  return (
+    <Modal size="xlarge" header="Download snippet" hideFooter closable {...props}>
+      <div className="flex flex-col items-start justify-between gap-4 px-6 py-3 mb-4">
+        <p className="text-sm mt-2">
+          You can download this snippet as a local{' '}
+          <a
+            className="text-brand"
+            href="https://supabase.com/docs/guides/cli/local-development#database-migrations"
+            target="_blank"
+            rel="noopener"
+          >
+            migration
+          </a>{' '}
+          file via the{' '}
+          <a
+            className="text-brand"
+            href="https://supabase.com/docs/guides/cli/getting-started"
+            target="_blank"
+            rel="noopener"
+          >
+            Supabase CLI
+          </a>
+          .
+        </p>
+        <Separator />
+      </div>
+
+      <TabsProvider>
+        <div className="flex flex-col items-start justify-between gap-4 px-6 py-3 relative mb-6">
+          <div className="absolute top-1 right-6">
+            <TwoOptionToggle
+              width={75}
+              options={['CLI', 'NPM']}
+              activeOption={selectedView}
+              borderOverride="border-gray-100"
+              onClickOption={() =>
+                selectedView === 'CLI' ? setSelectedView('NPM') : setSelectedView('CLI')
+              }
+            />
+          </div>
+
+          <Tabs type="underlined">
+            <Tabs.Panel id="migration" label="Migration">
+              <div className="flex flex-col gap-2 my-2 w-full">
+                <h2 className="text-lg">Download as migration</h2>
+                <p className="text-sm text-scale-1000">
+                  Use the snippet in a new migration named{' '}
```

Comment:
just wondering - should we also just state where to run this command in? help the user with less thinking

also, am not too sure what the outcome of this command is but is `use` the right verb here? should it be `download`? "Download the snippet into a new migration"?

Reply:
1. changed 'use' to 'download'. I think that's right. 
2. added links at the bottom here. Played with adding helper text above the tabs there as well, but felt like it was getting in the way. 
Do you think this is enough? The "learn more about" should convey:  "this is for local development", which may or may not be pertinent to you. Does it convey that well enough to you? 

<img width="870" alt="screenshot-2023-11-20-at-10 24 34" src="https://github.com/supabase/supabase/assets/105593/a0302d7a-c909-4c05-bc75-333c181e6f3e">

Reply:
![screenshot-2023-11-20-at-10 34 10](https://github.com/supabase/supabase/assets/105593/be28b873-9279-4153-89bf-ceb0c09f4b97)